### PR TITLE
fix(protocol-designer): fix touchtip mmFromEdge migration

### PR DESCRIPTION
**Pull request by ncdiehl11**
# Overview

Fix a bug where the default touchtip mmFromEdge (0mm) resulted in an empty input field and form errors. Refactors this default value to a constant. Also adds a check to form level errors to allow 0 value (previously emitted error for any falsy value).

Note— I am removing the touch tip speed required errors for now (the relevant fields will be migrated in a followup PR, and then we can add back those errors.)

## Test Plan and Hands on Testing

- import a legacy protocol with touchtip enabled on a moveLiquid step form 

- open a step form and verify that touch tip mm from edge value populates to `0` and no errors are shown

## Changelog

- update falsy value logic in `InputStepFormField`
- fix form level error checks for touchtip mm from edge
- temporarily remove touch tip speed required errors
- refactor default mm from edge to constant (`0` mm)

## Review requests

see test plan

## Risk assessment

low

### CODE REVIEW

**Review by jerader — APPROVED** [review]
lgtm!

File: protocol-designer/fixtures/protocol/8/doItAllV3MigratedToV8.json [hunk 1]
```diff
@@ -2753,31 +2753,262 @@
...
         "message": ""
       }
     },
+    {
+      "commandType": "pickUpTip",
+      "key": "14180e37-e219-43dd-bcfa-bd546a4f5786",
+      "params": {
+        "pipetteId": "0b3f2210-75c7-11ea-b42f-4b64e50f43e5",
+        "labwareId": "0b44c760-75c7-11ea-b42f-4b64e50f43e5:opentrons/opentrons_96_tiprack_300ul/1",
+        "wellName": "A1"
```
Comment (ddcc4) [inline protocol-designer/fixtures/protocol/8/doItAllV3MigratedToV8.json:2768]: Hm, what caused the extra new commands?

File: protocol-designer/src/components/molecules/InputStepFormField/index.tsx [hunk 2]
```diff
@@ -62,7 +62,7 @@ export function InputStepFormField(
             setIsPristine(false)
           }
         }}
-        value={value ? String(value) : null}
+        value={value?.toString()}
```
Comment (ddcc4) [inline protocol-designer/src/components/molecules/InputStepFormField/index.tsx:65]: For context, what **IS** this component here? Is it the textbox for the touch-tip distance, or something else?

File: protocol-designer/src/steplist/formLevel/errors.ts [hunk 3]
```diff
@@ -1138,15 +1138,19 @@ export const aspirateTouchTipMmFromEdgeRequired = (
   fields: HydratedMoveLiquidFormData
 ): FormError | null => {
   const { aspirate_touchTip_checkbox, aspirate_touchTip_mmFromEdge } = fields
-  return aspirate_touchTip_checkbox && !aspirate_touchTip_mmFromEdge
+  return aspirate_touchTip_checkbox &&
+    !aspirate_touchTip_mmFromEdge &&
+    aspirate_touchTip_mmFromEdge !== 0
```
Comment (ddcc4) [inline protocol-designer/src/steplist/formLevel/errors.ts:1143]: The extra `!aspirate_touchTip_mmFromEdge` is just to make TypeScript happy, right? (since it ought to be covered by `aspirate_touchTip_mmFromEdge !== 0`?)

**Review by ddcc4 — APPROVED** [review]

File: protocol-designer/src/components/molecules/InputStepFormField/index.tsx (hunk 2 above)
Comment (ncdiehl11) [reply protocol-designer/src/components/molecules/InputStepFormField/index.tsx:65]: `InputStepFormField` is a thin wrapper on our `InputField` component (adding some minor styling props and handling things like updating the form data when the input field text changes). In the UI, it corresponds to the text box along with title and caption
Screenshot 2025-04-21 at 8 45 29 AM

File: protocol-designer/src/steplist/formLevel/errors.ts (hunk 3 above)
Comment (ncdiehl11) [reply protocol-designer/src/steplist/formLevel/errors.ts:1143]: I think we need this additional condition because `0` is a valid value here. The falsy check `!aspirate_touchTip_mmFromEdge` checks for any falsy values, and `aspirate_touchTip_mmFromEdge !== 0` allows a valid `0` value

Comment (ddcc4) [reply protocol-designer/src/steplist/formLevel/errors.ts:1143]: Thanks -- my question was more like: would this code still be correct if we just deleted the `!aspirate_touchTip_mmFromEdge` condition?
